File: backend/Domain/Entities/ArchObject.cs

namespace Domain.Entities
{
    public class ArchObject : BaseLogDomain, IBaseDomain
    {
        public int Id { get; set; }
        public string? Address { get; set; }
        public string? Name { get; set; }
        public string? Identifier { get; set; }
        public int? DistrictId { get; set; }
        public string DistrictName { get; set; }
        public string? Description { get; set; }
        public double? XCoordinate { get; set; }
        public double? YCoordinate { get; set; }
    }
}

File: frontend/src/constants/ArchObject.ts
export type ArchObject = {
  id: number;
  address: string;
  name: string;
  identifier: string;
  district_id: number;
  xcoordinate: number;
  ycoordinate: number;
  description: string;
  name_kg: string;
  tags: number[];
  district_name?: string;
};


export interface ArchObjectValues extends ArchObject {
  geometry: any[];
  addressInfo: AddresInfo[];
  point: any[];
  DarekSearchList: [];
  errordistrict_id: string;
  errordescription: string;
  erroraddress: string;
  count?: number;
  legalRecords?:[]
  legalActs?:[]
}
export type AddresInfo = {
  address: string;
  propcode?: string;
}